pkg/tagger/collectors/workloadmeta_main.go

```diff
@@ -20,22 +20,15 @@ import (
 	"github.com/DataDog/datadog-agent/pkg/workloadmeta"
 )
 
-const (
-	workloadmetaCollectorName = "workloadmeta"
-
-	podSource       = workloadmetaCollectorName + "-" + string(workloadmeta.KindKubernetesPod)
-	taskSource      = workloadmetaCollectorName + "-" + string(workloadmeta.KindECSTask)
-	containerSource = workloadmetaCollectorName + "-" + string(workloadmeta.KindContainer)
-	gardenSource    = workloadmetaCollectorName + "-" + string(workloadmeta.KindGardenContainer)
-)
+type processor interface {
+	ProcessTagInfo([]*TagInfo)
+}
 
 // WorkloadMetaCollector collects tags from the metadata in the workloadmeta
 // store.
 type WorkloadMetaCollector struct {
-	store    workloadmeta.Store
-	children map[string]map[string]struct{}
-	out      chan<- []*TagInfo
-	stop     chan struct{}
+	store workloadmeta.Store
+	p     processor
```

Comment:
maybe worth to use the full name.

```suggestion
	processor     processor
```